### PR TITLE
Issue 88

### PR DESCRIPTION
**Pull request by jp-embedded**

### CODE REVIEW

File: src/cpp_output.cpp [hunk 1]
```diff
@@ -1126,7 +1126,8 @@ void cpp_output::gen_action_part_script(scxml_parser::action &a)
 	const string expr = a.attr["expr"];
 
 	//out << tab << tab << "// " << a.type << " expr=" << expr << endl;
-	out << tab << tab << expr << endl;
+	//out << tab << tab << expr << endl;
+	out << regex_replace(expr, regex("(^|\n)"), string("$1") + tab + tab) << endl;
```
Comment (sstiller) [inline src/cpp_output.cpp]: This modifies the code more than you might want.
```c++
std::cout << "Hello \
world!\n";
```
becomes 
```c++
		std::cout << "Hello \
		world!\n";
```
which modifies the string.

The user's code should not be touched, even if the indenting is not perfect.

Comment (jp-embedded) [reply src/cpp_output.cpp]: Thanks. You are right - I will change that back

Comment (jp-embedded) [reply src/cpp_output.cpp]: Of cause, lines ending with '\' could be skipped for indenting. But maybe there are other unexpected cases.

Comment (sstiller) [reply src/cpp_output.cpp]: Checking for `\` should work. I have no idea, where it would fail.

**Comment by jp-embedded** [comment]
upgraded msvc. regexp did not vork with old version

**Pull request merged**